lib/bolt/module_installer/puppetfile/forge_module.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'semantic_puppet'
+require 'set'
+
+require_relative '../../error'
+
+# This class represents a resolved Forge module.
+#
+module Bolt
+  class ModuleInstaller
+    class Puppetfile
+      class ForgeModule
+        NAME_REGEX    = %r{\A[a-z][a-z0-9_]*[-/](?<name>[a-z][a-z0-9_]*)\z}.freeze
+        REQUIRED_KEYS = Set.new(%w[name version]).freeze
+
+        attr_reader :full_name, :name, :semantic_version, :type, :version
+
+        def initialize(name, version)
+          @full_name, @name           = parse_name(name)
+          @version, @semantic_version = parse_version(version)
+          @type                       = :forge
+        end
+
+        def self.from_hash(hash)
+          new(hash['name'], hash['version'])
+        end
+
+        def self.implements?(hash)
+          REQUIRED_KEYS == hash.keys.to_set
+        end
+
+        # Formats the full name and extracts the module name.
+        #
+        private def parse_name(name)
+          unless (match = name.match(NAME_REGEX))
+            raise Bolt::ValidationError,
+                  "Invalid name for Forge module : #{name}. Name must match 'owner/name', "\
+                  "must start with a lowercase letter, and may only include lowercase "\
+                  "letters, digits, and underscores."
+          end
+
+          [name.tr('-', '/'), match[:name]]
+        end
+
+        # Parses the version into a Semantic Puppet version.
+        #
+        private def parse_version(version)
+          return unless version.is_a?(String)
+
+          unless SemanticPuppet::Version.valid?(version)
+            raise Bolt::ValidationError,
+                  "Invalid version for Forge module #{@full_name}: #{version.inspect}"
+          end
+
+          [version, SemanticPuppet::Version.parse(version)]
```

Comment:
Will 'version' ever be different from `SemanticPuppet::Version.parse(version).to_s`?

Reply:
No. Updated to just use a single `version` var.